src/pages/account/mijn-historiek/index.tsx

```diff
@@ -72,6 +78,24 @@ const AccountMyHistory: NextPage = () => {
 		[filters, setFilters]
 	);
 
+	const onClickRow = (visit: Visit) => {
+		getAccessStatus(visit.spaceId)
+			.then((response) => {
+				switch (response?.status) {
+					case VisitStatus.APPROVED:
+						router.push(`/${visit.spaceSlug}`);
+						break;
+					case VisitStatus.PENDING:
+						router.push(ROUTES.visitRequested.replace(':slug', visit.spaceSlug));
+						break;
+					default:
+						router.push(createHomeWithReadingRoomFilterUrl(visit));
+						break;
+				}
+			})
+			.catch(() => router.push(createHomeWithReadingRoomFilterUrl(visit)));
+	};
```

Comment:
Notify users of errors
Prefer the use of await (less nesting)

```suggestion
	const onClickRow = async (visit: Visit) => {
	try {
		 const response = await getAccessStatus(visit.spaceId)
		  switch (response?.status) {
			  case VisitStatus.APPROVED:
				  router.push(`/${visit.spaceSlug}`);
				  break;
			  case VisitStatus.PENDING:
				  router.push(ROUTES.visitRequested.replace(':slug', visit.spaceSlug));
				  break;
			  default:
				  router.push(createHomeWithReadingRoomFilterUrl(visit));
				  break;
		  }
	} catch (err) {
		console.error(err)
		toastService.notify ... // Het controleren van je toegang tot deze bezoekersruimte is mislukt.
		router.push(createHomeWithReadingRoomFilterUrl(visit)
	}
```

Reply:
Fixed 👍